app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
+
+  def saml_response_url
+    saml_request.response_url
+  end
```

Comment:
Isn't `saml_response_url` already defined by the `saml_idp` gem? We were calling it directly before without having to define it again.